### PR TITLE
feat(analysis): add ttlStrategy on AnalysisRun for garbage collecting stale AnalysisRun automatically

### PR DESCRIPTION
**Pull request by kevinqian-db**
This PR adds the following according to discussions in [Enhancement Proposal](https://github.com/argoproj/argo-rollouts/issues/3285). This is a first-stage PR that only contains change to AnalysisRun, but will have a second PR in the future for AnalysisTemplate. See more reasoning in the proposal above.

* Add `ttlStrategy` to AnalysisRun spec. It looks like
```
spec:
  ttlStrategy:
    secondsAfterCompletion: <number-of-seconds>
    secondsAfterFailure: <number-of-seconds>
    secondsAfterSuccess: <number-of-seconds>
```
Note that `secondsAfterSuccess` and `secondsAfterFailure` will override `secondsAfterCompletion` if specified, and the corresponding AnalysisRun is of the corresponding state.

* Add `completedAt` (next to `startedAt`) in AnalysisRun status.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore. [Enhancement Proposal](https://github.com/argoproj/argo-rollouts/issues/3285)
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by kevinqian-db** [comment]
@jessesuen @zachaller PTAL, thanks!

**Comment by zachaller** [comment]
Can you fix conflicts?

**Review by zachaller — CHANGES_REQUESTED** [review]
can we re-run codegen to fix conflicts

**Comment by kevinqian-db** [comment]
Rebased on master and reran codegen.

File: docs/features/analysis.md [hunk 1]
```diff
@@ -854,6 +854,24 @@ spec:
     limit: 20
 ```
 
+## Time-to-live (TTL) Strategy
+
+!!! important
+    Available since v1.6
```
Comment (zachaller) [inline docs/features/analysis.md:860]: This would be available in 1.7

File: pkg/apis/rollouts/v1alpha1/analysis_types.go [hunk 2]
```diff
@@ -134,6 +134,16 @@ type MeasurementRetention struct {
 	Limit int32 `json:"limit" protobuf:"varint,2,opt,name=limit"`
 }
 
+// TtlStrategy defines the strategy for the time to live depending on if the analysis succeeded or failed
+type TtlStrategy struct {
```
Comment (jessesuen) [inline pkg/apis/rollouts/v1alpha1/analysis_types.go:138]: NIT: can we adopt golang acronym casing for this type and name this `TTLStrategy` instead?

https://google.github.io/styleguide/go/decisions.html#initialisms

File: pkg/apis/rollouts/v1alpha1/analysis_types.go [hunk 3]
```diff
@@ -378,6 +388,9 @@ type AnalysisRunSpec struct {
 	// +patchStrategy=merge
 	// +optional
 	MeasurementRetention []MeasurementRetention `json:"measurementRetention,omitempty" patchStrategy:"merge" patchMergeKey:"metricName" protobuf:"bytes,5,rep,name=measurementRetention"`
+	// TtlStrategy object contains the strategy for the time to live depending on if the analysis succeeded or failed
+	// +optional
+	TtlStrategy *TtlStrategy `json:"ttlStrategy,omitempty" protobuf:"bytes,6,opt,name=ttlStrategy"`
```
Comment (jessesuen) [inline pkg/apis/rollouts/v1alpha1/analysis_types.go:393]: Same comment here, `TTLStrategy` for the field name.

File: docs/features/analysis.md (hunk 1 above)
Comment (kevinqian-db) [reply docs/features/analysis.md]: Done

File: pkg/apis/rollouts/v1alpha1/analysis_types.go (hunk 2 above)
Comment (kevinqian-db) [reply pkg/apis/rollouts/v1alpha1/analysis_types.go]: Done

File: pkg/apis/rollouts/v1alpha1/analysis_types.go (hunk 3 above)
Comment (kevinqian-db) [reply pkg/apis/rollouts/v1alpha1/analysis_types.go]: Done

**Review by zachaller — APPROVED** [review]

**Pull request merged**